packages/p2p/src/components/modal-manager/modals/filter-modal/filter-modal.jsx

```diff
@@ -39,156 +89,298 @@ const FilterModal = () => {
     };
 
-    React.useEffect(() => {
+    const onClickReset = () => {
+        buy_sell_store.onClickReset();
+        onClickClear();
+    };
+
+    const onClickApplyPaymentMethods = () => {
+        buy_sell_store.onClickApply(selected_methods, selected_methods_text);
         buy_sell_store.setShowFilterPaymentMethods(false);
-        my_profile_store.getPaymentMethodsList();
+    };
 
-        if (buy_sell_store.selected_payment_method_value.length) {
-            setSelectedMethods(buy_sell_store.selected_payment_method_value);
-            setSelectedMethodsText(buy_sell_store.selected_payment_method_text);
-        }
+    const onClickApply = () => {
+        buy_sell_store.onClickApply(selected_methods, selected_methods_text);
+        hideModal();
+    };
 
+    const diff = (arr1, arr2) => arr1.filter(x => !arr2.includes(x));
+    // if user has previously already selected some payment methods and clicked Apply
+    const has_already_selected_payment_methods =
+        buy_sell_store.selected_payment_method_value?.length &&
+        (selected_methods.length !== buy_sell_store.selected_payment_method_value.length ||
+            diff(selected_methods, buy_sell_store.selected_payment_method_value).length > 0);
+    // if user is selecting payment methods for the first time and has selected some payment methods
+    const has_recently_selected_payment_methods =
+        buy_sell_store.selected_payment_method_value?.length === 0 && selected_methods.length > 0;
+    // if user has previously selected all payment methods
+    const has_selected_all_payment_methods =
+        buy_sell_store.selected_payment_method_value.length === my_profile_store.payment_methods_list.length;
+    const has_selected_payment_methods = has_already_selected_payment_methods || has_recently_selected_payment_methods;
+
+    React.useEffect(() => {
+        my_profile_store.getPaymentMethodsList();
+
+        return () => {
+            my_profile_store.setSearchTerm('');
+            my_profile_store.setSearchResults([]);
+        };
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
-    const FilterModalResult = observer(() => {
-        if (buy_sell_store.is_filter_modal_loading) return <Loading is_fullscreen={false} />;
-        else if (my_profile_store.search_term) {
-            if (!my_profile_store.search_results || my_profile_store.search_results.length > 0) {
-                return my_profile_store.search_results?.map((payment_method, key) => {
-                    return (
-                        <Checkbox
-                            key={key}
-                            label={payment_method.text}
-                            onChange={() => onChange(payment_method)}
-                            value={selected_methods.includes(payment_method.value)}
-                        />
-                    );
-                });
-            }
-            return <FilterModalNoResults text={my_profile_store.search_term} />;
```

Comment:
All the extra spaces and removed lines here are the result of eslint after refactoring out some duplicated code into functions: https://github.com/binary-com/deriv-app/pull/7789/commits/ab0a8f23698be8d6f77640e433d23cce4cb113de?diff=split&w=0